.github/workflows/sycl-linux-run-tests.yml

```diff
@@ -284,11 +284,8 @@ jobs:
           echo "opts=$CMAKE_EXTRA_ARGS" >> $GITHUB_OUTPUT
         else
           if [ "${{ contains(inputs.target_devices, 'ext_oneapi_hip')  }}" == "true" ]; then
```

Comment:
I am of the opinion that we should move all the runner specific code to the caller instead. For post-commit, in sycl-post-commit: https://github.com/intel/llvm/blob/sycl/.github/workflows/sycl-post-commit.yml#L85, just like what we do for docker image options. In this case,  we can pass CMake build options to sycl-link-run-tests using `extra_cmake_args`.

Reply:
This isn't really runner specific code, it's target device specific code. Any time this workflow is called targeting `HIP` we will need this to run. Another downside is we would have to duplicate this code in the pre and postcommit callsites, or make another workflow that implements this script in call it in both places.

Reply:
My initial thoughts were the same as @uditagarwal97 suggested. I'm not sure if that duplication is enough to convince me to do things differently, but I don't have strong opinion about it anyway.